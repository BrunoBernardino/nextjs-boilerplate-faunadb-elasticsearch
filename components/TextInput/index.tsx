import React, { forwardRef } from 'react';
import styled from 'styled-components';

import { css } from 'lib/constants';

interface TextInputProps {
  name: string;
  placeholder?: string;
  isDisabled?: boolean;
  type?: 'text' | 'email' | 'password' | 'url' | 'search' | 'number';
  value: any;
  onChange?: any;
  onKeyDown?: any;
  onClick?: any;
  className?: string;
  ref?: any;
  id?: string;
}

const StyledInput = styled.input`
  color: ${css.textGray};
  font-size: 1.7rem;
  margin: 0.5rem 0 1rem;
  display: block;
  text-align: left;
  padding: 0.7rem 1.5rem;
  border-radius: 5px;
  font-weight: 500;
  background: ${css.white};
  transition: all 140ms ease-in;
  border: 1px solid ${css.borderGray};
  width: calc(100% - 3rem);

  &:hover,
  &:focus {
    border-color: ${css.pink};
    outline: none;
    box-shadow: 1px 1px 3px ${css.borderGray};
  }
`;

const TextInput: React.SFC<TextInputProps> = forwardRef(
  (props: TextInputProps, ref: any) => {
    const { id, name, className, isDisabled, ...remainingProps } = props;

    return (
      <StyledInput
        className={className}
        disabled={isDisabled}
        id={id || name}
        name={name}
        ref={ref}
        {...remainingProps}
      />
    );
  },
);

TextInput.defaultProps = {
  isDisabled: false,
  type: 'text',
};

export default TextInput;
