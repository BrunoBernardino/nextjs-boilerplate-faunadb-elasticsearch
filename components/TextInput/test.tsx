import React from 'react';
import renderer from 'react-test-renderer';
import { mount } from 'enzyme';
import expect from 'expect';
import enzymify from 'expect-enzyme';

import TextInput from './index';

expect.extend(enzymify());

describe('TextInput', () => {
  // NOTE: Skipping because enzyme doesn't work with React 17 yet
  it.skip('renders the text input without errors', () => {
    const wrapper = mount(
      <TextInput name="name" value="Bruno" onChange={() => {}} />,
    );
    expect(wrapper.find('input').props().value).toBe('Bruno');
    expect(wrapper.find('input').props().name).toBe('name');
  });

  it('renders the input as expected', () => {
    const tree = renderer
      .create(<TextInput name="name" value="Bruno" />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
