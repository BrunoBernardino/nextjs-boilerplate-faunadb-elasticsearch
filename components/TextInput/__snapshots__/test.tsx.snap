// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextInput renders the input as expected 1`] = `
<input
  className="TextInput__StyledInput-sc-1779p3-0 kaAhff"
  disabled={false}
  id="name"
  name="name"
  type="text"
  value="Bruno"
/>
`;
